packages/edit-site/src/components/header/document-actions/index.js

```diff
@@ -25,16 +25,41 @@ function useSecondaryText() {
 		selectedBlock,
 		getBlockParentsByBlockName,
 		getBlockWithoutInnerBlocks,
+		hoveredBlockIds,
+		getBlockName,
+		getBlock,
 	} = useSelect( ( select ) => {
+		const {
+			getSelectedBlock,
+			getHoveredBlocks,
+			getBlockParentsByBlockName: _getBlockParentsByBlockName,
+			__unstableGetBlockWithoutInnerBlocks,
+			getBlockName: _getBlockName,
+			getBlock: _getBlock,
+		} = select( 'core/block-editor' );
 		return {
-			selectedBlock: select( 'core/block-editor' ).getSelectedBlock(),
-			getBlockParentsByBlockName: select( 'core/block-editor' )
-				.getBlockParentsByBlockName,
-			getBlockWithoutInnerBlocks: select( 'core/block-editor' )
-				.__unstableGetBlockWithoutInnerBlocks,
+			selectedBlock: getSelectedBlock(),
+			getBlockParentsByBlockName: _getBlockParentsByBlockName,
+			getBlockWithoutInnerBlocks: __unstableGetBlockWithoutInnerBlocks,
+			hoveredBlockIds: getHoveredBlocks(),
+			getBlockName: _getBlockName,
+			getBlock: _getBlock,
 		};
 	} );
 
+	// Go through hovered blocks and see if one is of interest.
+	const hoveredTemplatePartBlockId = hoveredBlockIds.find(
```

Comment:
Maybe!  I don't think we have an immediate need for it, but it could be helpful for future use.

To be clear... do you mean something like `isBlockHovered( block.name )` or `isBlockHovered( clientId )`.  The second one is what makes sense to me, but from the context of the line you pinned this not on Im thinking you might mean the first?

Reply:
I think both could be helpful for different use cases, but the first is obviously more helpful to us here! Maybe `isBlockOfTypeHovered` or something more generic... I like the idea of adding it as a selector because it helps separate concerns a bit with this selector :)

Reply:
> the first is obviously more helpful to us here!

Thinking about it more Im not totally following.  Is there a need to retrieve a boolean for `is` one of that type hovered?  Or just to retrieve it if it is, like `getHoveredBlockByType` (we could use something like that here)?  I guess there are a handful of things that could potentially be useful, but does it make sense to add them all proactively or as we need them?

Reply:
maybe something like 6f3751e ?